src/execution_plan/ops/op_delete.c

```diff
@@ -177,6 +178,22 @@ static Record DeleteConsume(OpBase *opBase) {
 			array_append(op->deleted_edges, *e);
 			// If evaluating the expression allocated any memory, free it.
 			SIValue_Free(value);
+		} else if(type & T_PATH) {
+			Path *p = (Path *)value.ptrval;
+			size_t nodeCount = Path_NodeCount(p);
+
+			if(nodeCount > 0) {
+				Node *n = Path_GetNode(p, 0);
+				array_append(op->deleted_nodes, *n);
+
+				for(size_t i = 1; i < nodeCount; i++) {
+					Node *n = Path_GetNode(p, i);
+					array_append(op->deleted_nodes, *n);
+					Edge *e = Path_GetEdge(p, i - 1);
+					array_append(op->deleted_edges, *e);
+				}
```

Comment:
sorry after talking with @swilly22 we think it is better to split this to 2 loop to utilize the cpu cache better

```
for(size_t i = 0; i < nodeCount; i++) {
 	Node *n = Path_GetNode(p, i);
 	array_append(op->deleted_nodes, *n);
}
for(size_t i = 0; i < nodeCount - 1; i++) {
 	Edge *e = Path_GetEdge(p, i);
 	array_append(op->deleted_edges, *e);
}
```

Reply:
Ok. No problem. Change done